src/vs/platform/actions/browser/menuEntryActionViewItem.ts

```diff
@@ -313,4 +313,71 @@ export class LabeledMenuItemActionItem extends MenuEntryActionViewItem {
 		super.dispose();
 	}
 }
+export class LabeledNotebookMenuItemActionItem extends MenuEntryActionViewItem {
+	private _labeledItemClassDispose?: IDisposable;
+
+	constructor(
+		public _action: MenuItemAction,
+		@IKeybindingService labeledkeybindingService: IKeybindingService,
+		@INotificationService protected _notificationService: INotificationService,
+		private readonly _defaultCSSClassToAdd: string = ''
+	) {
+		super(_action, labeledkeybindingService, _notificationService);
+	}
+
+	updateLabel(): void {
+		if (this.label) {
+			this.label.innerText = this._commandAction.label;
+		}
+	}
+
+	// Overwrite item class to ensure that we can pass in a CSS class that other items use
+	// Leverages the _defaultCSSClassToAdd property that's passed into the constructor
+	protected _updateItemClass(item: ICommandAction): void {
+		dispose(this._labeledItemClassDispose);
+		this._labeledItemClassDispose = undefined;
+
+		if (ThemeIcon.isThemeIcon(item.icon)) {
+			// TODO
+		} else if (item.icon) {
+			let iconClass: string;
+
+
+			if (item.icon?.dark?.scheme) {
+				const iconPathMapKey = item.icon.dark.toString();
+
+				if (ICON_PATH_TO_CSS_RULES.has(iconPathMapKey)) {
+					iconClass = ICON_PATH_TO_CSS_RULES.get(iconPathMapKey)!;
+				} else {
+					iconClass = ids.nextId();
+					createCSSRule(`.codicon.masked-icon.${iconClass}::before`, `-webkit-mask-image: ${asCSSUrl(item.icon.light || item.icon.dark)}`);
```

Comment:
Why aren't you adding the `mask-image` style as well like all the other rules?

Reply:
Added. Just so you know - ADS does not recognize the style rule without the -webkit- prefix. That said, I'll add the non-prefixed rule for consistency sake -- this is something found also in common-icons.css; and it was done to future-proof the styles in case ADS is ever rendered with a non-Chromium engine.